app/controllers/home_controller.rb

```diff
@@ -1,6 +1,20 @@
 # frozen_string_literal: true
 
 class HomeController < ApplicationController
+  before_action :eligable?, only: %i(setting profile)
+
   def index
   end
+
+  def profile
+  end
+
+  def setting
+  end
+
+  private
+
+  def eligable?
```

Comment:
this is a good decision, but it's better to change it a little bit. Let's use gem Action policy for such things. I will add description to task